### DISCUSSION TITLE
Erro ao tentar gravar valores com casas decimais "is not a valid floating point value"

### DISCUSSION THREAD

**Discussion by avelsys** [HashLoad/horse#257] [Q&A] [answered]
Boa noite pessoal estou tentando passar um valor com casas decimais utilizando o Horse e DataSet.Serialize, mais não estou conseguindo. Estou utilizando o banco postgres o campo salario esta como numeric(12,2),  mais toda vez que tento gravar ele dá esse erro: is not a valid floating point value

Alguem poderia me ajudar neste sentido?

{
    "cli_nome":"Alessandro Avelino ",
    "cli_endereco":"Rua x",
    "cli_bairro":"",
    "cli_salario":1500.45,
    "cli_dtnascimento":"1975-07-12"
}

**Comment by viniciussanchez** [comment, marked as answer]
Primeiramente o problema não está no Horse. Deve ser alguma coisa no DataSet-Serialize. Como que você está fazendo? Os fields está adicionado no seu DataSet? Ou está deixando eles sendo criados de forma automática pelo DataSet-Serialize? Está usando a última versão disponível dos componentes envolvidos?

**Reply by gabrielbaltazar** [reply]
Como o @viniciussanchez comentou não é no Horse. Provavelmente é no DataSet-Serialize na hora de fazer o parse. Acredito que você tenha mudado recentemente para o Delphi 11, eu também tive esse problema numa lib de parse de JSON e contornei tratando numa config de FormatSettings. Não sei se no DataSet-Serialize tem alguma config pra isso e se ele já está tratando esse caso também. Seguindo a dica do @viniciussanchez, verifique se suas libs estão atualizadas e se o DataSet-Serialize está contornando esse caso do Delphi 11.

**Reply by avelsys** [reply]
Boa noite! Obrigado pelo Feedback pessoal!! Realmente no Delphi 11 precisei alterar o FormatSettings, fiz a alteração para pegar o '.', mais fiz pelas configurações do DataSet.Serialize! Dessa forma aqui: 

TDataSetSerializeConfig.GetInstance.Import.DecimalSeparator := '.';

Rodou sem problemas! Obrigado pessoal!
